daemon/algod/api/server/v1/handlers/handlers.go

```diff
@@ -55,6 +55,16 @@ func nodeStatus(node *node.AlgorandFullNode) (res v1.NodeStatus, err error) {
 	}, nil
 }
 
+func txEncode(tx transactions.Transaction, ad transactions.ApplyData) (res v1.Transaction, err error) {
+	if tx.Type == protocol.PaymentTx {
```

Comment:
Use a case statement here to switch between the different transaction types.

Reply:
To be honest @tsachiherman , I'm a beginner in Golang (i'm a C/Assembler guy) and, because they are strings, I did with multiple ifs.

Reply:
https://play.golang.org/p/9nre4pHnfwn

Reply:
Sorry for the delay (pending work to do). Done